ui/src/app/cohort-search/list-search-v2/list-search-v2.component.tsx

```diff
@@ -336,6 +340,11 @@ export const ListSearchV2 = fp.flow(withCdrVersions(), withCurrentWorkspace(), w
       return [Domain.CONDITION, Domain.PROCEDURE].includes(this.props.searchContext.domain);
     }
 
+    selectIconDisabled() {
+      const {selectedIds, source} = this.props;
+      return source !== 'criteria' && selectedIds && selectedIds.length === 1000;
```

Comment:
Can you put a similar check in `CriteriaTree` component to limit selections in the hierarchy view also? I was able to max it out, then switch to the tree and select more concepts. If the layout works, it may be better to move the message up to `CriteriaSearch` so we can use the same one for both views.

Reply:
Also, can you change`===` to `>=` so it will disable in case somehow we get more more than 1000 selections? 
And should we remove `source !== 'criteria'` since I think we need this limit fo cohort builder also? We'd just need to show different text in the message if `source === 'criteria'`.

Reply:
I was not able to check the tree structure for concept Set. Regarding cohort i will create another PR just to keep them separate.